src/components/UserDetailSections/UserAffiliations/UserAffiliations.js

```diff
@@ -46,10 +51,41 @@ const UserAffiliations = ({
 
   const isLoading = isFetching || isAffiliationsMutating;
 
-  const onUpdateAffiliations = useCallback(({ added, removed }) => {
-    return handleAssignment({ added, removed })
-      .then(refetch);
-  }, [handleAssignment, refetch]);
+  const onUpdateAffiliations = useCallback(async ({ added, removed }) => {
+    try {
+      const responses = await handleAssignment({ added, removed });
+      const errors = await getResponseErrors(responses);
+
+      if (errors.length) {
+        callout.sendCallout({
+          type: 'error',
+          message: (
+            <div>
+              <div>
+                <strong>
+                  <FormattedMessage id="ui-users.affiliations.manager.modal.changes.error" />
+                </strong>
+              </div>
+              <ul className={css.errorsList}>
+                {errors.map(({ message }) => <li key={message}>{message}</li>)}
```

Comment:
I believe we have to display already formatted messages based on error codes instead of the messages from the response itself

Reply:
My question has not been answered by PO yet. I will add them if they provide the translation.